### DISCUSSION TITLE
Trigger sync via Kubernetes API

### DISCUSSION THREAD

**Discussion by vonbox** [argoproj/argo-cd#20797] [Q&A]
I have ArgoCD running in a Kubernetes cluster and I want to trigger an Application sync from within a pod. If I can do that via the Kubernetes API, by POSTing or PUTting something to an Application resource, I would not need to worry about authenticating to the ArgoCD API. Is that possible.
In other words; when the ArgoCD API server receives a request to do a sync, does is handle that by calling the Kubernetes API server to update the state of the Application resource, or does it somehow give the Application controller a direct instruction. If the first, what does that update look like?

**Comment by m8rmclaren** [comment]
I also have this question

**Reply by m8rmclaren** [reply]
Just kidding - you can actually trigger a sync like this by setting the Operation on the `Application` object. ArgoCD automatically removes this when the sync is complete.

```go
import argocdv1alpha1 "github.com/argoproj/argo-cd/v3/pkg/apis/application/v1alpha1"

app := object.(*argocdv1alpha1.Application)
app.Operation = &argocdv1alpha1.Operation{
	Sync: &argocdv1alpha1.SyncOperation{
		Revision: "<HEAD/etc.>",
	},
}

```
